python/cudf/cudf/core/column/column.py

```diff
@@ -1468,26 +1360,26 @@ def kernel_non_masked(values, results):
                 results[i] = core(values[i])
 
         kernel_non_masked.forall(len(column))(values, results)
-    # Output
-    return Buffer(results)
+
+    return as_column(results)
 
 
-def _data_from_cuda_array_interface_desc(desc):
+def _data_from_cuda_array_interface_desc(obj):
+    desc = obj.__cuda_array_interface__
     ptr = desc["data"][0]
     nelem = desc["shape"][0]
     dtype = np.dtype(desc["typestr"])
 
-    data = rmm.device_array_from_ptr(
-        ptr, nelem=nelem, dtype=dtype, finalizer=None
-    )
-    data = Buffer(data)
+    # TODO: this can be done more efficiently
```

Comment:
Think we can drop this comment as it refers to code that has already been removed.

```suggestion
```